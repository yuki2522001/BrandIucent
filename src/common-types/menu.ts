export interface MenuType {
  key: string;
  label: string;
  url: string;
}

export interface MenuProps {
  menuList: MenuType[];
}