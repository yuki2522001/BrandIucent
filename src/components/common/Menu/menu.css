/* css for component menu */
.menu__list {
  display: flex;
  align-items: center;
}

.menu__item {
  list-style: none;
  cursor: pointer;
  font-size: 16px;
  line-height: 19px;
  margin-right: 32px;
}

.menu__item > a {
  text-decoration: none;
  font-weight: 400;
  color: var(--dark);
}

/* css for component menuContact */
.contact__item {
  margin-top: 20px;
}

.contact__item > a {
  text-decoration: none;
  color: var(--light);
  font-size: 15px;
  line-height: 20px;
}

.contact__item > a:hover {
  color: var(--text-secondary);
}