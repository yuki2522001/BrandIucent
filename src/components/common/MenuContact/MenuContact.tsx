import React, { memo } from "react";
import { MenuProps, MenuType } from "@common-types/menu";

const MenuContact: React.FC<MenuProps> = memo(({ menuList }) => {
  const renderContactList = (list: MenuType[]) => {
    return list.map((item) => (
      <li className="contact__item" key={item.key}>
        <a href={item.url}>{item.label}</a>
      </li>
    ));
  }

  return (
    <div data-testid="contact" className="contact">
      <ul className="contact__list">
        {renderContactList(menuList)}
      </ul>
    </div>
  );
});

export default MenuContact;