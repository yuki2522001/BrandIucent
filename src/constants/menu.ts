import { MenuType } from "../common-types/menu";

const MENU_LIST: MenuType[] = [
  {
    key: "orders",
    label: "Orders",
    url: "/",
  },
  {
    key: "cart",
    label: "Cart",
    url: "/",
  },
  {
    key: "contact",
    label: "Contact",
    url: "/",
  },
];

const MENU_SERVICE: MenuType[] = [
  {
    key: "email-marketing",
    label: "Email Marketing",
    url: "/",
  },
  {
    key: "campaigns",
    label: "Campaigns",
    url: "/",
  },
  {
    key: "branding",
    label: "Branding",
    url: "/",
  },
  {
    key: "offline",
    label: "Offline",
    url: "/",
  },
];

const MENU_ABOUT: MenuType[] = [
  {
    key: "our-story",
    label: "Our Story",
    url: "/",
  },
  {
    key: "benefits",
    label: "Benefits",
    url: "/",
  },
  {
    key: "team",
    label: "Team",
    url: "/",
  },
  {
    key: "careers",
    label: "Careers",
    url: "/",
  },
];

const MENU_HELP: MenuType[] = [
  {
    key: "FAQs",
    label: "FAQs",
    url: "/",
  },
  {
    key: "contact-us",
    label: "Contact Us",
    url: "/",
  },
];

export { MENU_LIST, MENU_SERVICE, MENU_ABOUT, MENU_HELP };
