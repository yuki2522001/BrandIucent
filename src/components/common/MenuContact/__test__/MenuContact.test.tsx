import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import MenuContact from "../MenuContact";
import { MENU_SERVICE } from "@constants/menu";

describe("MenuContact component", () => {
  test("should render MenuContact component", () => {
    const { getByTestId } = render(<MenuContact menuList={MENU_SERVICE} />);
    expect(getByTestId("contact")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<MenuContact menuList={MENU_SERVICE} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
