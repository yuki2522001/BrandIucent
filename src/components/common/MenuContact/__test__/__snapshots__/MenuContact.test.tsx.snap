// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuContact component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="contact"
    data-testid="contact"
  >
    <ul
      class="contact__list"
    >
      <li
        class="contact__item"
      >
        <a
          href="/"
        >
          Email Marketing
        </a>
      </li>
      <li
        class="contact__item"
      >
        <a
          href="/"
        >
          Campaigns
        </a>
      </li>
      <li
        class="contact__item"
      >
        <a
          href="/"
        >
          Branding
        </a>
      </li>
      <li
        class="contact__item"
      >
        <a
          href="/"
        >
          Offline
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
